executor/aggregate.go

```diff
@@ -680,35 +675,34 @@ func (e *HashAggExec) execute(ctx context.Context) (err error) {
 	}
 }
 
-func (e *HashAggExec) getGroupKey(row chunk.Row) ([]byte, error) {
-	vals := make([]types.Datum, 0, len(e.GroupByItems))
+func (e *HashAggExec) getGroupKey(row chunk.Row) (string, error) {
```

Comment:
This function can be further optimized. Maybe in the next PR.

Reply:
we can change the way to calculate the `[]byte` according to the `group by` items.

Reply:
this method maybe better return `[]byte` directly and make to string lazy happen, and then modify `StringSet` add a `Exists` override method that accept `[]byte` as argument, then use this small trick https://github.com/golang/go/commit/8072f46abdf5b2d3ed0ee7d691823b7fcdaa7c21#diff-590553aade2ea7ef319660a8151c5eb9R581

I feel @coocood want to make `StringSet` be `map[[n]byte]struct` if we have some "magic design" make groupKey be fixed length? :cat:

Reply:
maybe `hack.String` or `hack.Slice` is what you want.